v2/tools/generator/internal/config/object_model_configuration.go

```diff
@@ -222,3 +283,34 @@ func (omc *ObjectModelConfiguration) configuredGroups() []string {
 
 	return result
 }
+
+// CreateTestObjectModelConfiguration builds up a new configuration for a particular type, returning both the top level
+// configuration and the type configuration.
+// While intended only for test use, this isn't in a _test.go file as we want to use it from tests in multiple packages.
```

Comment:
IMO it's awkward having code only for test alongside "production" code. Would putting this into the `testcommon` package work? That package already exists, is used in multiple test packages, and isn't next to the production code.

There's quite a bit of shared stuff there such as the test gomega matchers, test resoruce namer, and test context stuff which feels like it goes well with this?

Reply:
I agree with your concerns. That said, I put this in the same package so it could have access to private fields, not sure about moving it. Will consider, but not in this PR.